src/System.CommandLine/Parsing/ArgumentResult.cs

```diff
@@ -83,7 +83,7 @@ public void OnlyTake(int numberOfTokens)
 
                 if (!string.IsNullOrWhiteSpace(ErrorMessage))
                 {
-                    return new ParseError(ErrorMessage!, this);
+                    return new ParseError(ErrorMessage!, Parent is OptionResult option ? option : this);
```

Comment:
This change is not obvious. Before this change, the `SymbolResult` of `ParseError` for an Argument with unmatched tokens owned by an Option was the `OptionResult`, not `ArgumentResult`:

https://github.com/dotnet/command-line-api/blob/0adeb26f4ce52c454280d9f3a5d6975e5d081fec/src/System.CommandLine/Parsing/ParseResultVisitor.cs#L509

but this was true only "non-custom errors" (user defined, coming from validators):

https://github.com/dotnet/command-line-api/blob/0adeb26f4ce52c454280d9f3a5d6975e5d081fec/src/System.CommandLine/Parsing/ParseResultVisitor.cs#L510

Without this change, one test was failing:

```log
  Failed System.CommandLine.Tests.ParseDiagramTests.Parse_result_diagram_displays_unmatched_tokens [283 ms]
  Error Message:
   Expected string to be
"[ command ![ -x <ar> ] ]", but
"[ command [ -x !<ar> ] ]" differs near "[ -" (index 10).
```

Since we hide the `Option.Argument` from the user, I think it's reasonable to use the `OptionResult` in such cases for all validations.